source/material_model/compositing.cc

```diff
@@ -0,0 +1,207 @@
+                                 "property is not one of the valid values."));
+      return Property::viscosity;
+      return Property::property_map.at(s);
+    }
+
+    //Copy the requested data for one model
+    template <int dim>
+    void
+    Compositing<dim>::copy_required_properties(const unsigned int model_index,
+                                               const typename Interface<dim>::MaterialModelOutputs &base_output,
+                                               typename Interface<dim>::MaterialModelOutputs &out) const
+    {
+      if (model_property_map.at(Property::viscosity) == model_index)
+        out.viscosities = base_output.viscosities;
+      if (model_property_map.at(Property::density) == model_index)
+        out.densities = base_output.densities;
+      if (model_property_map.at(Property::thermal_expansion_coefficient) == model_index)
+        out.thermal_expansion_coefficients = base_output.thermal_expansion_coefficients;
+      if (model_property_map.at(Property::specific_heat) == model_index)
+        out.specific_heat = base_output.specific_heat;
+      if (model_property_map.at(Property::thermal_conductivity) == model_index)
+        out.thermal_conductivities = base_output.thermal_conductivities;
+      if (model_property_map.at(Property::compressibility) == model_index)
+        out.compressibilities = base_output.compressibilities;
+      if (model_property_map.at(Property::entropy_derivative_pressure) == model_index)
+        out.entropy_derivative_pressure = base_output.entropy_derivative_pressure;
+      if (model_property_map.at(Property::entropy_derivative_temperature) == model_index)
+        out.entropy_derivative_temperature = base_output.entropy_derivative_temperature;
+      if (model_property_map.at(Property::reaction_terms) == model_index)
+        out.reaction_terms = base_output.reaction_terms;
+    }
+
+    template <int dim>
+    void
+    Compositing<dim>::evaluate(const typename Interface<dim>::MaterialModelInputs &in,
+                               typename Interface<dim>::MaterialModelOutputs &out) const
+    {
+      typename Interface<dim>::MaterialModelOutputs base_output(out.viscosities.size(),
+                                                                this->introspection().n_compositional_fields);
+
+      for (unsigned int i=0; i<models.size(); ++i)
+        {
+          models[i]->evaluate(in, base_output);
+          copy_required_properties(i, base_output, out);
+        }
+    }
+
+    template <int dim>
+    void
+    Compositing<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Material model");
+      {
+        prm.enter_subsection("Compositing");
+        {
+          std::map<std::string, Property::MaterialProperty>::const_iterator prop_it = Property::property_map.begin();
+          for (; prop_it != Property::property_map.end(); ++prop_it)
+            {
+              prm.declare_entry(prop_it->first, "compositing",
```

Comment:
I suppose you chose `compositing` as entry because it is invalid? In other places we use a value called `unspecified`, and then attach this value to the list of valid model patterns in the line below as: `MaterialModel::get_valid_model_names_pattern<dim>()+"|unspecified"`